docs/configuration.rst

```diff
@@ -28,10 +28,49 @@ Top level keys
 
 Custom fragment types
 ---------------------
+``Towncrier`` allows defining custom fragment types. Custom fragment types
+will be used instead ``towncrier`` default ones, they are not combined.
 
-Custom fragment types can be including in the
-pyproject.toml.
-Each custom type (``[[tool.towncrier.type]]``) has the following keys:
+
+Custom fragment types can be including in the pyproject.toml in two different ways:
+
+Defining custom fragment types with toml tables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Users can create their own custom fragment types by adding tables to
+the pyproject.toml named ``[tool.towncrier.type.<a custom fragment type>]``.
+
+These tables may include the following optional keys:
+
+ * ``name``: The description of the fragment type, as it must be included
+   in the news file. If omitted, it defaults to  its  fragment type,
+   but capitalized.
+ * ``showcontent``: Whether if the fragment contents should be included in the
+   news file. If omitted, it defaults to ``true``
+
+
+For example:
+
+.. code-block:: toml
+
+
+        [tool.towncrier]
+        [tool.towncrier.type.chores]
```

Comment:
can we have this like `[tool.towncrier.chores]`  with the mention that `type` is a reserved word.

I think this would look more "natural"

I don't like at all the old syntax like the one before...maybe it should be deprecated and suggestion new users to use the new syntax ? 

```
[[tool.towncrier.type]]
directory = "chore"
```

Thanks!

Reply:
Sure, no problem. 

I would like to confirm something. If using named tables like `[tool.towncrier.chores]` or `tool.towncrier.feat]` 
they will be mixed with other fields like `package`, `package_dir`, and so on. 

If mappings  are reserved to custom types, then I can get anything that behaves like a mapping and parse the fragment type from it.

An alternative that I like is adding a new field to the [tool.towncrier] table (for example, `custom_fragment_types`) that contains a list with all the custom types. This could be interesting since with default values, you would not need to add a named table per each fragment type, but for the ones that you want to configure. For example:
```
[tool.towncrier]
project="bar"

custom_fragment_types = [
  "feat",
  "refactor",  
  "fix", 
  "doc", 
  "chore"
]

[tool.towncrier.chores]
       name = "Other Tasks"
        showcontent = False
```

What do you think @adiroiban ?

Reply:
I don't know if this should be named `custom_fragment_types` ... maybe just `available_types` or `fragment_types`  ?

I have not used TOML to much so I don't know what are the best practices for TOML.

We can also keep `[tool.towncrier.type.chores]` as maybe for TOML users this is normal.

----------

I don't care that much about this.

I set up towncrier for my project in 2017 and never had to change it since then :)

Reply:
I have remove `type` from the table name for the new way to configure fragment types (the existing one still uses `[[tool.towncrier.type]]`). So we can have `[tool.towncrier.chores]` or `[tool.towncrier.feat]`. 

But I can change  it to something like `[tool.towncrier.fragment.chores]` if you prefer. That would prevent conflicts between custom fragment types and other fields of `[tool.towncrier]`.